src/dbnode/persist/schema/types.go

```diff
@@ -56,12 +56,14 @@ type IndexBloomFilterInfo struct {
 
 // IndexEntry stores entry-level data indexing
 type IndexEntry struct {
-	Index       int64
-	ID          []byte
-	Size        int64
-	Offset      int64
-	Checksum    int64
-	EncodedTags []byte
+	Index        int64
+	ID           []byte
+	Size         int64
+	Offset       int64
+	DataChecksum int64
+	EncodedTags  []byte
+	// NOTE: add any new fields before IndexEntryChecksum. Decoding assumes checksum is final bytes of IndexEntry
+	IndexEntryChecksum int64
```

Comment:
It could be a little weird that we expose this on the struct, but end up calculating it implicitly in the encoder (i.e. ignoring whatever the value is set to when passed in). If we end up doing the validation in the decoder, we could probably just remove this altogether from the struct here and just add a comment or something to that effect to alert folks to it's existence in the index file.

Reply:
Hm yeah, I'd probably be in favor of mentioning that it exists as part of a comment on the struct (either at the top as part of existing comment for `IndexEntry` or inline here as a comment, either way works well I leave that to your best judgement).  Thoughts?

Reply:
Yeah, think this is the right approach. Will update